### ISSUE TITLE
Error raised when work with php-cs-fixer

### ISSUE THREAD

**Issue by chrisyue** [phpactor/phpactor#2202]
The use of php-cs-fixer as a formatter is not working. after I turned logging.enabled on, I got this log message in application.log:

```
[2023-04-13 14:35:14] phpactor.ERROR: Diagnostic error from provider "Phpactor\Extension\LanguageServerPhpCsFixer\Provider\PhpCsFixerDiagnosticsProvider": php-cs-fixer exited with code '127'; cmd: '/Users/chris/Repositories/api/vendor/bin/php-cs-fixer' 'fix' '--dry-run' '--verbose' '--format' 'json' '-'; stderr: 'env: php: No such file or directory '; stdout: '' {"channel":"LSPDIAG","trace":"#0 [internal function]: Phpactor\\Extension\\LanguageServerPhpCsFixer\\Model\\PhpCsFixerProcess->Phpactor\\Extension\\LanguageServerPhpCsFixer\\Model\\{closure}()\n#1 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Coroutine.php(118): Generator->send(127)\n#2 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Internal/Placeholder.php(149): Amp\\Coroutine->Amp\\{closure}(NULL, '')\n#3 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Coroutine.php(123): Amp\\Coroutine->resolve('')\n#4 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Internal/Placeholder.php(149): Amp\\Coroutine->Amp\\{closure}(NULL, NULL)\n#5 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Deferred.php(53): Amp\\Promise@anonymous->resolve(NULL)\n#6 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/byte-stream/lib/ResourceInputStream.php(101): Amp\\Deferred->resolve(NULL)\n#7 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Loop/NativeDriver.php(327): Amp\\ByteStream\\ResourceInputStream::Amp\\ByteStream\\{closure}('ai', Resource id #2852, NULL)\n#8 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Loop/NativeDriver.php(124): Amp\\Loop\\NativeDriver->selectStreams(Array, Array, 0.058)\n#9 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Loop/Driver.php(138): Amp\\Loop\\NativeDriver->dispatch(true)\n#10 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Loop/Driver.php(72): Amp\\Loop\\Driver->tick()\n#11 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/amphp/amp/lib/Loop.php(95): Amp\\Loop\\Driver->run()\n#12 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/phpactor/language-server/lib/Core/Server/LanguageServer.php(110): Amp\\Loop::run(Object(Closure))\n#13 /Users/chris/.local/share/nvim/mason/packages/phpactor/lib/Extension/LanguageServer/Command/StartCommand.php(50): Phpactor\\LanguageServer\\Core\\Server\\LanguageServer->run()\n#14 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/symfony/console/Command/Command.php(298): Phpactor\\Extension\\LanguageServer\\Command\\StartCommand->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#15 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/symfony/console/Application.php(1040): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#16 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/symfony/console/Application.php(301): Symfony\\Component\\Console\\Application->doRunCommand(Object(Phpactor\\Extension\\LanguageServer\\Command\\StartCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#17 /Users/chris/.local/share/nvim/mason/packages/phpactor/lib/Application.php(47): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#18 /Users/chris/.local/share/nvim/mason/packages/phpactor/vendor/symfony/console/Application.php(171): Phpactor\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#19 /Users/chris/.local/share/nvim/mason/packages/phpactor/bin/phpactor(45): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#20 {main}"} []
```

It seems like the php bin can't be found when running php-cs-fixer, but it's working well if I run php-cs-fixer directly in terminal. it's strange that if the php is not found, neither can phpactor be run.

**Comment by chrisyue** [comment]
Yes it does

```shell
$ /Users/chris/Repositories/api/vendor/bin/php-cs-fixer fix --dry-run
Loaded config default from "/Users/chris/Repositories/api/.php-cs-fixer.dist.php".
Using cache file ".php-cs-fixer.cache".
   1) src/Entity/DragonTreasure.php

Found 1 of 7 files that can be fixed in 0.029 seconds, 14.000 MB memory used
```

**Comment by chrisyue** [comment]
PHP

```
php -v
PHP 8.2.4 (cli) (built: Mar 16 2023 16:13:06) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.4, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.4, Copyright (c), by Zend Technologies
```

installed by homebrew

```
which php
/opt/homebrew/bin/php
```

Neovim

```
nvim --version
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.9.0/share/nvim"
```

no docker.

And sorry I don't know what wsdl is.

**Comment by dantleech** [comment]
what about `env php` ?

**Comment by chrisyue** [comment]
`env php` is OK.

I think there is nothing wrong with the environment, most likely.
If there's something wrong with the env or PHP, we could never see this error message, because the error itself is raised by PHP, right?

Maybe there are some ways I could help debugging?

**Comment by dantleech** [comment]
It has to be the way the process is Started from Phpactor.  You can check the (PhpCsFixerProcess?) class and mess about with that.

**Comment by chrisyue** [comment]
I've just figured it out why this error occured, please check the PR.

Besides passing the PATH env to the process, we also could pass the absolute path of  PHP bin to process, before `$this->binPath`.

**Comment by dantleech** [comment]
Yes, specifying the full path my be better. Will check to see of we need to do it in other places

**Comment by dantleech** [comment]
Thanks for investigating this and finding the solution, I fixed it in #2211 if you can confirm, I'll close this ticket in anticipation that it fixes it :)
